decimal.go

```diff
@@ -469,13 +469,30 @@ func (d Decimal) MarshalJSON() ([]byte, error) {
 
 // Scan implements the sql.Scanner interface for database deserialization.
 func (d *Decimal) Scan(value interface{}) error {
-	str, err := unquoteIfQuoted(value)
-	if err != nil {
+	var err error
+	// first try to see if the data is stored in database as a Numeric datatype
+	switch value.(type) {
+
+	case float64:
+		// numeric in sqlite3 sends us float64
+		*d = NewFromFloat(value.(float64))
+		return err
+
+	case int64:
+		// at least in sqlite3 when the value is 0 in db, the data is sent
+		// to us as an int64 instead of a float64 ...
+		*d = NewFromFloat(float64(value.(int64)))
```

Comment:
This will lose precision if the value is indeed an int64. float64 only has 52 bits of precision, so it can't exactly represent every 64 bit integer. You should create the decimal directly from the int64, by calling `New(value, 0)`.